### PR TITLE
Add more explanation for -segment-ov parameter

### PR DESCRIPTION
**Pull request by tszumski**
# Description
Add more explanation for -segment-ov parameter
# Issue
#545 

# Author(s)
@tszumski

# Performance impact

- [ ] quality
- [ ] memory
- [ ] speed
- [ ] 8 bit
- [ ] 10 bit
- [x] N/A

# Merge method
- [x] Allow the maintainer to squash and merge when PR is ready to create a 1-commit to the master branch. The maintainer will be able to fix typos / combine commit messages to create a more readable 1-commit message or use whatever is stated in the 'Description' section
- [ ] I will clean up my commits and the maintainer shall use 'rebase and merge' to the master branch

### CODE REVIEW

File: Config/SVTSegmentOvFile.txt [hunk 1]
```diff
@@ -2,14 +2,43 @@
...
+# 1. Purpose/use case of this feature
+#    This feature is rather for very specific use-case like surveillance systems, where objects like pedestrians or vehicles are detected and tracked (encoded with higher quality) while background is encoded with worse quality.
+#    Thanks to encoding only key areas with good quality bitstream is smaller and still have high value
+# 2. Usage:
+# For each frame in video user can specify regions with better or worse quality.
 # One input frame have to be preceded by the number of regions
 # Each region consist of 6 numbers (X1,Y1,X2,Y2, data, mode) i.e (X1,Y1)top left and (X2,Y2)bottom right corner
+# Region is translated into subset of LCU(largest coding unit) for which user override default qp
 # Mode parameter have to be in range [1..5]. This allow user to specify what to override by the [data]
-#    1.   Set qp value for each region, [data] is clipped to [0:51]
```
Comment (intelmark) [inline Config/SVTSegmentOvFile.txt:8]: # Mode 1   Set the qp value for each region. [data] is clipped to [0:51] .
# Mode 2   Change the qp value (i.e increase or decrease by [data]). [data] is clipped to [-25:25] ,
# Mode 3   Modify not only the quality but also blocks splitting(LCU partition). The behaviour depends on parameter 1 and 2 and the 5th parameter is unused.
# Mode 4   Only modify the block's splitting(LCU partition) (i.e increase or decrease by[data]). [data] is clipped to [-25:25].  Negative value increase block division.
# Mode 5   Update first stage of quantization in range [-7;7] positive value reduce bitrate, negative increase  bitrate.
#
# There are several ways to use mode 3.  You can set:
#    0 0 500 500 35 1 // Set new qp value
#    0 0 500 500 35 3 // Also modify the block splitting.  The 5th value is skipped
#    or
#    0 0 500 500 -5 2 // Increase the block quality by 5
#    0 0 500 500 -5 3 // Also modify block splitting.  The 5th value is skipped
#
# 3. Example:
#    If you want to increase the quality by 10 of an object of size 100x100 which is moving from left to right on the screen, you can use:
#    1    // one region for frame 0
#    0 0 100 100 -10 2
#    1    // one region for frame 1
#    10 0 110 100 -10 2
#    1    // one region for frame 2
#    20 0 120 100 -10 2
#    1    // one region for frame 3
#    30 0 130 100 -10 2
#    1    // one region for frame 4
#    40 0 140 100 -10 2
#    1    // one region for frame 5
#    50 0 150 100 -10 2
#    1    // one region for frame 6
#    60 0 160 100 -10 2**

**Comment by tszumski** [comment]
Hi @intelmark, Thanks for review and comments. I updated description with your suggestions.

File: Config/SVTSegmentOvFile.txt [hunk 2]
```diff
@@ -2,14 +2,44 @@
...
+#    30 0 130 100 -10 2
+#    1    // one region for frame 4
+#    40 0 140 100 -10 2
+#    1    // one region for frame 5
+#    50 0 150 100 -10 2
+#    1    // one region for frame 6
+#    60 0 160 100 -10 2
 3
 0 0 500 500 50 1
 500 500 1000 0 -20 4
```
Comment (tianjunwork) [inline Config/SVTSegmentOvFile.txt:45]: I don't quite understand the example here.
```
3
0 0 500 500 50 1
500 500 1000 0 -20 4
500 500 1000 1000 20 4
2
0 0 500 500 20 1
0 0 300 200 20 1
```
Do it mean
3 regions for frame 0, which is `[0,0] - [500, 500]`, `[500,500] - [1000, 0] is this valid region?`, `[500,500] - [1000,1000]`
2 regions for frame 1, which is `[0,0] - [500, 500]`,  `[0,0] - [300,200]`. why they are overlapped?

What is the rule for the rest of the frames?

Comment (tszumski) [reply Config/SVTSegmentOvFile.txt:45]: Hi @tianjunwork 
> 3 regions for frame 0, which is [0,0] - [500, 500], [500,500] - [1000, 0] is this valid region?, [500,500] - [1000,1000]
Correct, 3 regions, all of them are valid
> 2 regions for frame 1, which is [0,0] - [500, 500], [0,0] - [300,200]. why they are overlapped?
They are overlapped to show that this kind of action is allowed

Comment (tianjunwork) [reply Config/SVTSegmentOvFile.txt:45]: `(X1,Y1)is the top left and (X2,Y2) is the bottom right corner` [500,500] - [1000, 0] is this valid region?
Could you add `#` line in this example too to explain that region can be overlapped?
Rule for each frames needs to be specified?  In this example, only frame 0 and 1 have rule?  What is the rule for the rest of the frames?

Comment (tszumski) [reply Config/SVTSegmentOvFile.txt:45]: > (X1,Y1)is the top left and (X2,Y2) is the bottom right corner [500,500] - [1000, 0] is this valid region?

This is general description, the most important thing is that points in space have to be diagonal to create rectangle

> Rule for each frames needs to be specified? In this example, only frame 0 and 1 have rule? What is the rule for the rest of the frames?

If the number of frames > number of "rules" the application will loop to the beginning of the file

File: Config/SVTSegmentOvFile.txt [hunk 3]
```diff
@@ -2,14 +2,47 @@
...
-#    3.   Use this mode to modify not only quality but also blocks splitting, behaviour depends on parameter 1 and 2
-#    4.   Use this mode to modify blocks splitting i.e increase or decrease by the [data] clipped to [-25:25], negative value increase block division
-#    5.   Update first stage of quantization in range [-7;7] positive value reduce bitrate, negative increase
+# 1. Purpose/use case of this feature
+#    This feature is for very specific use-cases like surveillance systems, where dynamic objects such as pedestrians or vehicles are detected and tracked (encoded with higher quality) while the static background is encoded with worse quality.
+#    By encoding only the important/key areas with good quality, the resulting bitstream is smaller and still has high usefulness.
+# 2. Usage:
+# For each frame in the video, the user can specify regions with better or worse quality.
+# One input frame have to be preceded by the number of regions.
+# Each region consists of 6 numbers (X1, Y1, X2, Y2, data, mode) i.e (X1,Y1)is the top left and (X2,Y2) is the bottom right corner.
```
Comment (intelmark) [inline Config/SVTSegmentOvFile.txt:11]: Each region consists of 6 numbers (X1, Y1, X2, Y2, data, mode) i.e (X1,Y1) is the top left and (X2,Y2) is the bottom right corner.  Regions are allowed to overlap in the same frame.

**Comment by tianjunwork** [comment]
Hi @tszumski , you can push incrementally, so that we can review what is changed. If there is no need to use force push, it should be avoid.

**Comment by intelmark** [comment]
Current SVT-HEVC user guide needs some updating:
Path to segment override file, diffrence between largest and smallest value should be less or equal 25. Note: using this feature will disable ImproveSharpness and BitRateReduction parameters

maybe

_Path to segment override file which will allow for sharpness improvement and bit rate reduction on a per segment basis.  Refer to config/SVTSegmentOvFile.txt for details._

**Pull request merged**